### ISSUE TITLE
Post-training quantization

### ISSUE THREAD

**Issue by David-LETINAUD** [tensorflow/tensorflow#40831]
Hi everyone,

I've recently worked with the 2 models: "conv" and "low_latency_conv" of the speech_commands example.
```
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/speech_commands
```
I succeed to obtain respectively, 88% and 81% of testing accuracy.
I'd like to use a quantization of int16 (int8 will be just as good), but I can't because of the tf.contrib.quantize function which is deprecated in tf2.0. It advises me to install tf1.15, unfortunately, I'm not allowed to install it.

I've tried this solution which seems interesting:
```
converter = tf.lite.TFLiteConverter.from_saved_model(saved_model_dir)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_quant_model = converter.convert()
```

With a pbtxt saved model, the "from_saved_model" function crash with the error :
```
OSError: Cannot parse file saved_model.pbtxt': 1:1 : Message type "tensorflow.SavedModel" has no field named "node"..
```
And when I convert it to a frozen model (with freeze.py) to have a "save_model.pb", it crashes in"converter.convert()"
with the error:
```
"ValueError: This converter can only convert a single ConcreteFunction. Converting multiple functions is under development."
```

Has anyone found a way to quantize it ?

P.S.: In the description of the create_conv_model function (in models.py), I think the second layer [MaxPool] is missing in the code.

**System information**
- Windows 7
- TensorFlow 2.1.0

**Comment by lheim** [comment]
Hi @David-LETINAUD,
I'm not sure if understand your question correctly. Are you trying to save your just converted model?

As seen in your code, you're using Tensorflow Lite for this. You can save your `tflite_quant_model` by simply:
```python
open('tflite_quant_model.tflite', 'wb').write(tflite_quant_model)
```

You can then load this model with the Tensorflow Lite Interpreter, see more [here](https://www.tensorflow.org/lite/guide/inference#load_and_run_a_model_in_python).
For more about the quantization process, see [here](https://www.tensorflow.org/lite/performance/post_training_quantization).

**Comment by David-LETINAUD** [comment]
Hi @lheim 
I've already trained a speech recognition model with tensorflow 2.1. It gives me saved_model.pbtxt or saved_model.pb (if i convert it to a frozen model). 
I want to go a bit deeper and try to find the loss of accuracy between a model that stores weights in 32bits and another one with 16bits (or even 8bits).
Since I can't quantify it during training (tf 1.15 missing), I tried to find a way to quantify it after training. 
On the internet, I found a way to do this with tflite, which opens the tensorflow model to convert it back to a tflite model, quantify it, and then save it.

This is the little script, which is supposed to do just that:

```
import tensorflow as tf

saved_model_dir = "/model_path/"

converter = tf.lite.TFLiteConverter.from_saved_model(saved_model_dir)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
tflite_quant_model = converter.convert()
```
